src/nginx/t/init_service_configs_multiple.t

```diff
@@ -0,0 +1,294 @@
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+# OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+# SUCH DAMAGE.
+#
+################################################################################
+#
+use strict;
+use warnings;
+use JSON::PP;
+
+################################################################################
+
+use src::nginx::t::ApiManager;    # Must be first (sets up import path to
+                                  # the Nginx test module)
+use src::nginx::t::HttpServer;
+use src::nginx::t::ServiceControl;
+use Test::Nginx;    # Imports Nginx's test module
+use Test::More;     # And the test framework
+
+################################################################################
+
+# Port assignments
+my $NginxPort          = ApiManager::pick_port();
+my $BackendPort        = ApiManager::pick_port();
+my $ServiceControlPort = ApiManager::pick_port();
+
+my $t = Test::Nginx->new()->has(qw/http proxy/)->plan(27);
+
+# Save servce configuration that disables the report cache.
+# Report request will be sent for each client request
+$t->write_file('server.pb.txt', <<"EOF");
+service_control_config {
+  report_aggregator_config {
+    cache_entries: 0
+    flush_interval_ms: 1000
+  }
+}
+init_service_configs {
+  service_config_file_full_path: "service.1.pb.txt"
+  traffic_percentage: 80
+}
+init_service_configs {
+  service_config_file_full_path: "service.2.pb.txt"
+  traffic_percentage: 20
+}
+EOF
+
+# Save service name in the service configuration protocol buffer file.
+$t->write_file( 'service.1.pb.txt',
+  ApiManager::get_bookstore_service_config . <<"EOF");
+control {
+  environment: "http://127.0.0.1:${ServiceControlPort}"
+}
+quota {
+ metric_rules [
+   {
+     selector: "ListShelves"
+     metric_costs: [
+       {
+         key: "metrics_first"
+         value: 2
+       },
+       {
+         key: "metrics_second"
+         value: 1
+       }
+     ]
+   }
+ ]
+}
+EOF
+
+# Save service name in the service configuration protocol buffer file.
+$t->write_file( 'service.2.pb.txt',
+  ApiManager::get_bookstore_service_config_2 . <<"EOF");
```

Comment:
what is the different between 1 and 2?    If just config_id change,  can you use script to update it and generate a new file?